test/react_test_utils_test.dart

```diff
@@ -0,0 +1,213 @@
+    test('input', () => testEvent(Simulate.input));
+    test('keyDown', () => testEvent(Simulate.keyDown));
+    test('keyPress', () => testEvent(Simulate.keyPress));
+    test('keyUp', () => testEvent(Simulate.keyUp));
+    test('mouseDown', () => testEvent(Simulate.mouseDown));
+    test('mouseMove', () => testEvent(Simulate.mouseMove));
+    test('mouseOut', () => testEvent(Simulate.mouseOut));
+    test('mouseOver', () => testEvent(Simulate.mouseOver));
+    test('mouseUp', () => testEvent(Simulate.mouseUp));
+    test('paste', () => testEvent(Simulate.paste));
+    test('scroll', () => testEvent(Simulate.scroll));
+    test('submit', () => testEvent(Simulate.submit));
+    test('touchCancel', () => testEvent(Simulate.touchCancel));
+    test('touchEnd', () => testEvent(Simulate.touchEnd));
+    test('touchMove', () => testEvent(Simulate.touchMove));
+    test('touchStart', () => testEvent(Simulate.touchStart));
+    test('wheel', () => testEvent(Simulate.wheel));
+  });
+
+  test('findRenderedDOMComponentWithClass', () {
+    component = renderIntoDocument(sampleComponent({}));
+    var spanComponent = findRenderedDOMComponentWithClass(component, 'span1');
+
+    expect(spanComponent['tagName'], equals('SPAN'));
+  });
+
+  test('findRenderedDOMComponentWithTag', () {
+    component = renderIntoDocument(sampleComponent({}));
+    var h1Component = findRenderedDOMComponentWithTag(component, 'h1');
+
+    expect(h1Component['tagName'], equals('H1'));
+  });
+
+  test('findRenderedComponentWithType', () {
+    component = renderIntoDocument(div({}, [sampleComponent({})]));
+    var result = findRenderedComponentWithType(component, sampleComponent);
+    expect(isCompositeComponentWithType(result, sampleComponent), isTrue);
+  });
+
+  group('isCompositeComponent', () {
+    test('returns true when element is a React component', () {
+      component = renderIntoDocument(eventComponent({}));
+
+      expect(isCompositeComponent(component), isTrue);
+    });
+
+    test('returns false when element is not a React component', () {
+      component = renderIntoDocument(div({}));
+
+      expect(isCompositeComponent(component), isFalse);
+    });
+  });
+
+  group('isCompositeComponentWithType', () {
+    var renderedInstance = renderIntoDocument(sampleComponent({}));
+
+    test('returns true when element is a React component with class', () {
+      expect(isCompositeComponentWithType(
+          renderedInstance, sampleComponent), isTrue);
+    });
+
+    test('returns false when element is not a React component with class', () {
+      expect(isCompositeComponentWithType(
+          renderedInstance, eventComponent), isFalse);
+    });
+  });
+
+  group('isDOMComponent', () {
+    test('returns true when argument is a DOM component', () {
+      component = renderIntoDocument(sampleComponent({}));
+      var h1Element = findRenderedDOMComponentWithTag(component, 'h1');
+
+      expect(isDOMComponent(h1Element), isTrue);
+    });
```

Comment:
This test could be simplified:
```dart
     test('returns true when argument is a DOM component', () {
      component = renderIntoDocument(h1({}));
      expect(isDOMComponent(component), isTrue);
     });
```
Doesn't make too much of a difference, though.